marshmallow/schema.py

```diff
@@ -99,6 +101,37 @@ def __new__(mcs, name, bases, attrs):
         for attr in mcs.FUNC_LISTS:
             attr_copy = copy.copy(getattr(klass, attr))
             setattr(klass, attr, attr_copy)
+
+        # Now add in the decorated processors. By doing this after constructing
+        # the class, we let standard inheritance do all the hard work.
+        mro = klass.mro()
+
+        klass.processors = defaultdict(list)
+        for attr_name in dir(klass):
```

Comment:
Agreed that users *should* handle order by putting ordered operations in the same callback. But they might not, and if they don't, I could imagine this being a debugging headache. Anyway, if we don't want callbacks to run in any particular order, the docs should be clear about that, so that users don't make assumptions and get confused.